File: caesar.erl
% Kevin Pietrow

-module(caesar).
-export([uppercase/1, encrypt2/2, encrypt/2, decrypt2/2, decrypt/2, solve/2]).

uppercase([L|T]) -> 
	if 
		L > 96 ->
			[(L - 32)|uppercase(T)];
		true ->
			[L|uppercase(T)]
	end;
uppercase([]) -> [].

encrypt2([H|T], X) ->
	if 
		H == 32 ->
			[H|encrypt2(T, X)];
		H + X > 90 ->
			[(64 + (H + X - 90))|encrypt2(T, X)];
		true ->
			[(H + X)|encrypt2(T, X)]
	end;
encrypt2([], X) -> [].

encrypt(L, X) -> encrypt2(uppercase(L), X).			
		
decrypt2([H|T], X) ->
	if 
		H == 32 ->
			[H|decrypt2(T, X)];
		H - X < 65 ->
			[(91 - (65 - (H - X)))|decrypt2(T, X)];
		true ->
			[(H - X)|decrypt2(T, X)]
	end;
decrypt2([], X) -> [].

decrypt(L, X) -> decrypt2(uppercase(L), X).			
		
solve(L, X) ->
	if
		X >= 0 ->
			string:concat(decrypt(L, X), string:concat(" | ", solve(L, (X - 1))));
		true ->
			[]
	end.
	

File: caesar.scala
/** Kevin Pietrow */

object caesar {
	def encrypt(input2:String, num: Int) = {
		var input = input2.toUpperCase();
		
		var temp = "";
		
		var i = 0;
	
		for( i <- 0 until (input.length) ) {
			if (input(i).toInt == 32) {
				temp = temp + ((32).toChar).toString;
			}
		
			else if((input(i).toInt + num) > 90) {
				temp = temp + ((64 + (input(i).toInt + num - 90)).toChar).toString;
			}
			
			else {
				temp = temp + ((input(i).toInt + num).toChar).toString;
			}
		}
		println(temp);
	}

	def decrypt(input2:String, num: Int) = {
		var input = input2.toUpperCase();
		
		var temp = "";
		
		var i = 0;
		for( i <- 0 until (input.length) ) {
			if (input(i).toInt == 32) {
				temp = temp + ((32).toChar).toString;
			}
			
			else if((input(i).toInt - num) < 65) {
				temp = temp + ((91 - (65 - (input(i).toInt - num))).toChar).toString;
			}
			
			else {
				temp = temp + ((input(i).toInt - num).toChar).toString;
			}
		}
		
		println(temp);
	} 
	
	def solve(input:String, num: Int) {
			for( i <- (num) to 0 by -1) {
				decrypt(input, i);
			}
		
	}
		


		def main(args: Array[String]) {
			println("Encrypt: with \"Scala time\" as the string, and 17 as the shift.");
			encrypt("Scala Time", 17);
			println("\nDecrypt: with \"Get to THE Chopper\" as the string, and 6 as the shift.");
			decrypt("Get to THE Chopper", 6);
			println("\nSolve: with \"ScaLA in the house JA\" as the string, and the max shift as 26.");
			solve("ScaLA in the house JA", 26);
		}
}
